src/pages/BlogDetail.jsx

```diff
@@ -41,8 +21,8 @@ function BlogDetail() {
   };
 
   useEffect(() => {
-    fetchPost(postId);
-  }, [postId]);
+    setPost(fetchPost(postId));
+  }, [fetchPost, postId]);
```

Comment:
This only has to run when `postId` changes. `fetchPost` is a function. Functions never change. They produce output when called. `postId` is a state, and it changes.